src/Compilation.zig

```diff
@@ -2621,7 +2625,10 @@ fn updateStage1Module(comp: *Compilation, main_progress_node: *std.Progress.Node
     const prev_hash_state = man.hash.peekBin();
     const input_file_count = man.files.items.len;
 
-    if (try man.hit()) {
+    if (man.hit() catch |err| {
+        const filename = man.files.items[main_file_index].path orelse main_zig_file;
+        return CompilationResult{ .HashError = try CompilationResult.HashError.init(comp.gpa, filename, @errorName(err)) };
```

Comment:
This could be greatly simplified by calling `fatal` here but then the issue would that we don't know if it is the `main_zig_file` that caused the error or one of the files added by `--pkg-begin`.